scripts/pdm-lock.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# Default directories list
+default_directories=(
+    "."
+    "backend"
+    "prompt-service"
+    "worker"
+    "unstract/core"
+    "unstract/flags"
+    "platform-service"
+    "x2text-service"
+    "unstract/connectors"
+    "unstract/tool-sandbox"
+)
+
+# Check if directories are passed as arguments, otherwise use the default list
+if [[ $# -eq 0 ]]; then
+    directories=("${default_directories[@]}")
+else
+    directories=("$@")
+fi
+
+for dir in "${directories[@]}"; do
+    file_path="$dir/pyproject.toml"
+
+    # Check if the pyproject.toml file exists
+    if [[ ! -f "$file_path" ]]; then
+        echo "No pyproject.toml found in $dir"
+        continue  # Skip to the next iteration if the file does not exist
+    fi
+
+    # Check if there are changes in pyproject.toml against the main branch
+    git fetch origin main
```

Comment:
@chandrasekharan-zipstack I think we can do this outside loop. We need to do this only one time